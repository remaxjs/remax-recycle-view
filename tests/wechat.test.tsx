import React from 'react';
import Remax from 'remax';
import AppContainer from 'remax/esm/AppContainer';
import RecycleView from '../src/index';

jest.mock('remax', () => {
  return { ...jest.requireActual('remax'), Platform: { current: 'wechat' } };
});

describe('src/index.tsx', () => {
  it('should return correct LIST_HEIGHT in wechat', () => {
    const container = new AppContainer({});
    Remax.render(
      <RecycleView data={[{ height: 100 }]} renderItem={() => <div>hello</div>} />,
      container,
    );
    expect(container.root.toJSON()).toMatchSnapshot();
  });
});
