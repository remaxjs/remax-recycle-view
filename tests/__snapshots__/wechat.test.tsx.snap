// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index.tsx should return correct LIST_HEIGHT in wechat 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "text": "hello world.",
          "type": "plain-text",
        },
      ],
      "id": 2,
      "props": Object {
        "lowerThreshold": 50,
        "onScroll": "$$REMAX_METHOD_2_onScroll",
        "scrollTop": undefined,
        "style": "height:100rpx;",
        "upperThreshold": 50,
      },
      "text": undefined,
      "type": "scroll-view",
    },
  ],
  "id": 0,
  "props": undefined,
  "text": undefined,
  "type": "root",
}
`;
